File: erlang/raindrops/src/raindrops.erl
-module(raindrops).

-export([convert/1]).

convert(_Number) when _Number rem 3 =/= 0, _Number rem 5 =/= 0, _Number rem 7 =/= 0 -> integer_to_list(_Number);
convert(_Number) -> pling(_Number) ++ plang(_Number) ++ plong(_Number).

pling(_Number) when _Number rem 3 == 0 -> "Pling";
pling(_Number) -> "".

plang(_Number) when _Number rem 5 == 0 -> "Plang";
plang(_Number) -> "".

plong(_Number) when _Number rem 7 == 0 -> "Plong";
plong(_Number) -> "".

File: csharp/raindrops/Raindrops.cs
using System;

public static class Raindrops
{
    public static string Convert(int number)
    {
        string pling = Plxng(number, 3, "Pling");
        string plang = Plxng(number, 5, "Plang");
        string plong = Plxng(number, 7, "Plong");
        string result = pling + plang + plong;

        return result.Length == 0 ? number.ToString() : result;
    }

    private static string Plxng(int number, int div, string sound) {
        return number % div == 0 ? sound : "";
    }

}